### PR TITLE
Adds missing assertions and reorganizes them

### LINKED ISSUE #686: [Bug]: Some public `tm_` module parameters are not checked and unordered
Some parameters of module functions are not validated with `checkmate`:

Here's a survey _(that may not be complete)_:

- all that have this: `pre_output`, `post_output`
  - They are assumed to be validated by `{teal.widgets}` functions
- `tm_g_bivariate`: `facet`
- `tm_g_distribution`: `plot_height`, `plot_width`
- `tm_outliers`: `plot_height`, `plot_width`

I propose we couple this with:

- Reorder assertions with same parameter order
- Adds some comments to increase readability of initial common logic
  - `# Normalizing data types`
  - `# Start of assertions`
  - `# End of assertions`

Example:

```r
tm_g_foo <- function(label = "Regression Analysis",
                     regressor,
                     response,
                     shape = shape_names,
                     ggtheme = c("gray", "bw", "linedraw", "light", "dark", "minimal", "classic", "void")) {
  logger::log_info("Initializing tm_g_foo")

  # Normalize the parameters
  if (inherits(regressor, "data_extract_spec")) regressor <- list(regressor)
  if (inherits(response, "data_extract_spec")) response <- list(response)

  # Start of assertions
  checkmate::assert_string(label)
  checkmate::assert_list(regressor, types = "data_extract_spec")
  checkmate::assert_list(response, types = "data_extract_spec")
  if (!all(vapply(response, function(x) !(x$select$multiple), logical(1)))) {
    stop("'response' should not allow multiple selection")
  }

  checkmate::assert_character(shape)
  ggtheme <- match.arg(ggtheme)
  # End of assertions

  # ...
```

### PR DESCRIPTION
**Pull request by averissimo**
# Pull Request

- Fixes #686
- Part of #624 

#### Changes descriptions

- Adds missing assertions described on #686 
- Adds a helper function called `assert_single_selection`
  - Avoids many repetitive calls across functions to check for multiple selections
  - Adds simple unit tests
- Adds some comments to maintain a consistent structure

### CODE REVIEW

File: R/tm_missing_data.R [hunk 1]
```diff
@@ -87,29 +87,41 @@ tm_missing_data <- function(label = "Missing data",
...
+  logger::log_info("Initializing tm_missing_data")
+
+  # Requires Suggested packages
   if (!requireNamespace("gridExtra", quietly = TRUE)) {
     stop("Cannot load gridExtra - please install the package or restart your session.")
   }
   if (!requireNamespace("rlang", quietly = TRUE)) {
     stop("Cannot load rlang - please install the package or restart your session.")
   }
-  logger::log_info("Initializing tm_missing_data")
```
Comment (gogonzo) [inline R/tm_missing_data.R:96]: I'm fascinated by randomly placed logger entries

**Review by gogonzo — COMMENTED** [review]
Buddy, you've forgotten `tm_a_regression`. Few minor comments beside this

File: R/tm_file_viewer.R [hunk 2]
```diff
@@ -39,25 +39,24 @@
...
     checkmate::check_character(input_path, min.len = 1)
   )
-
   if (length(input_path) > 0) {
     valid_url <- function(url_input, timeout = 2) {
       con <- try(url(url_input), silent = TRUE)
       check <- suppressWarnings(try(open.connection(con, open = "rt", timeout = timeout), silent = TRUE)[1])
       try(close.connection(con), silent = TRUE)
-      ifelse(is.null(check), TRUE, FALSE)
+      is.null(check)
```
[the remark this answers is not in the document]
Comment (chlebowa) [reply R/tm_file_viewer.R:59]: :laughing:

[the remark this answers is not in the document]
Comment (chlebowa) [reply R/tm_file_viewer.R:59]: This is your brain on Python :wink:

File: tests/testthat/test-utils.R [hunk 3]
```diff
@@ -0,0 +1,22 @@
+testthat::test_that("assert_single_selection will pass if all elements have multiple choices disabled", {
```
[the remark this answers is not in the document]
Comment (averissimo) [reply tests/testthat/test-utils.R:1]: I know, I know, but this package is not well-tested enough to do it via public functions.

[the remark this answers is not in the document]
Comment (gogonzo) [reply tests/testthat/test-utils.R:1]: This increase coverage but problem stays the same. Feel free to merge these tests as-is and have a sort of a protection for these functions to not change them accidentally in a future. When the time comes to write proper tests I will bring this back 👍

[the remark this answers is not in the document]
Comment (averissimo) [reply tests/testthat/test-utils.R]: I commented that, but that excuse was lame :sweat_smile: 

I've updated the test to use `tm_g_bivariate` to test it out with minimal specs.

File: R/tm_g_bivariate.R [hunk 4]
```diff
@@ -265,6 +252,28 @@ tm_g_bivariate <- function(label = "Bivariate Plots",
...
+  checkmate::assert_numeric(plot_height, len = 3, any.missing = FALSE, finite = TRUE)
+  checkmate::assert_numeric(plot_height[1], lower = plot_height[2], upper = plot_height[3], .var.name = "plot_height")
+  checkmate::assert_numeric(plot_width, len = 3, any.missing = FALSE, null.ok = TRUE, finite = TRUE)
+  checkmate::assert_numeric(
+    plot_width[1],
+    lower = plot_width[2], upper = plot_width[3], null.ok = TRUE, .var.name = "plot_width"
+  )
+
+  checkmate::assert_flag(rotate_xaxis_labels)
+  checkmate::assert_flag(swap_axes)
```
[the remark this answers is not in the document]
Comment (averissimo) [reply R/tm_g_bivariate.R:266]: Done

File: R/tm_a_pca.R [hunk 5]
```diff
@@ -132,25 +156,11 @@ tm_a_pca <- function(label = "Principal Component Analysis",
...
-  )
-
-  plot_choices <- c("Elbow plot", "Circle plot", "Biplot", "Eigenvector plot")
-  checkmate::assert_list(ggplot2_args, types = "ggplot2_args")
-  checkmate::assert_subset(names(ggplot2_args), c("default", plot_choices))
+  checkmate::assert_multi_class(pre_output, c("shiny.tag", "shiny.tag.list", "html"), null.ok = TRUE)
+  checkmate::assert_multi_class(post_output, c("shiny.tag", "shiny.tag.list", "html"), null.ok = TRUE)
+  # End of assertions
 
+  # Send ui args
```
[the remark this answers is not in the document]
Comment (averissimo) [reply R/tm_a_pca.R]: Done

**Comment by averissimo** [comment]
`tm_a_regression` was already done upstream on `pre-release@main` branch. Now merged here and updated with `Send/Make UI args`

File: tests/testthat/test-utils.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (averissimo) [reply tests/testthat/test-utils.R]: @gogonzo What do you think about it as it is?

File: tests/testthat/test-utils.R [hunk 6]
```diff
@@ -1,14 +1,44 @@
...
+  local_logger_threshold(logger::FATAL)
+
   mock_spec <- data_extract_spec(
     dataname = "MOCK_DATASET",
     select = teal.transform::select_spec(choices = c("A", "B"), multiple = FALSE)
   )
 
-  testthat::expect_true(assert_single_selection(list(mock_spec)))
-  testthat::expect_true(assert_single_selection(list(mock_spec, mock_spec)))
+  testthat::expect_s3_class(
```
[the remark this answers is not in the document]
Comment (averissimo) [reply tests/testthat/test-utils.R]: Done.

Added some tests that will cover the new function and all the logic in the module creation.

[the remark this answers is not in the document]
Comment (averissimo) [reply tests/testthat/test-utils.R]: Also adds a superficial check on UI and server functions

File: tests/testthat/test-tm_g_bivariate.R [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+    mock_data_extract_spec(select_multiple = FALSE),
+    mock_data_extract_spec(select_multiple = FALSE)
+  )
+
+  testthat::expect_contains(
+    names(mod$ui_args),
+    names(formals(tm_g_bivariate))
+  )
+})
+
```
Comment (gogonzo) [inline tests/testthat/test-tm_g_bivariate.R:67]: I think you can skip this. `ui_args` contains too many elements - not sure if even half off them are really used in the ui.

```suggestion

```

File: tests/testthat/test-tm_g_bivariate.R [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+testthat::test_that("tm_g_bivariate creates a module with `ui_args` that have all arguments of function", {
```
Comment (gogonzo) [inline tests/testthat/test-tm_g_bivariate.R:68]:
```suggestion
testthat::test_that("tm_g_bivariate creates a module with datanames taken from data extracts", {
```

File: tests/testthat/test-tm_g_bivariate.R [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+    args = c(mod$server_args, data = shiny::reactive(data)),
+    expr = {
+      checkmate::expect_data_frame(
+        anl_merged_q()[["ANL"]],
+        min.rows = 1
+      )
+    }
+  )
+})
+
```
Comment (gogonzo) [inline tests/testthat/test-tm_g_bivariate.R:147]: Please skip testing ui and server - this should be asserted in a `teal::module`
```suggestion
```

**Review by gogonzo — APPROVED** [review]
Assertions are fine. I would remove few tests to avoid opening a new pandora box discussion

**Comment by m7pr** [comment]
You are true legends guys. Great work

**Pull request merged**